.github/workflows/gating.yaml

```diff
@@ -39,7 +41,7 @@ jobs:
       run: |
         pip3 install --upgrade pip
         pip3 install --upgrade setuptools
-        pip3 install --upgrade coveralls==3.2.0
+        pip3 install --upgrade coveralls
```

Comment:
I will take another look at this.

Reply:
Well, I got an error with pinned version:

 `coverage.misc.CoverageException: Can't read 'pyproject.toml' without TOML support. Install with [toml] extra`

Reply:
> Well, I got an error with the pinned version:
>
> `coverage.misc.CoverageException: Can't read 'pyproject.toml' without TOML support. Install with [toml] extra`

This has nothing to do with pinning/not pinning the version though. The reason why you're seeing the error is caused by your own change in this PR:
```
diff --git a/.github/workflows/gating.yaml b/.github/workflows/gating.yaml
index 3ade3e65..1b9bf7a6 100644
--- a/.github/workflows/gating.yaml
+++ b/.github/workflows/gating.yaml
@@ -28,7 +28,9 @@ jobs:
     - name: Install dependencies
       run: |
         python -m pip install --upgrade pip
-        pip install tox tox-gh-actions
+        pip install tox
+        pip install tox-gh-actions
+        pip install coverage
```

In the pre-coveralls step you install the latest coverage there is, which is >=7, however, then you install an unpinned version of `coveralls` which, since it has an upper boundary on coverage<7 it will downgrade it to coverage==6.X [which moved away from `setup.cfg` to `pyproject.toml`](https://github.com/nedbat/coveragepy/releases/tag/6.5.0) and so now it requires you to install the extra toml dependency [it has](https://github.com/nedbat/coveragepy/blob/3d57a072ec6073988b4c775a9cffd431ef63ab5f/setup.py#L100). Moreover, if you look at our `requirements-extras.txt` it's exactly what we're doing already for local dev/test envs: 
```
requirements-extras.txt=258=coverage[toml]==7.4.3 \
requirements-extras.txt:311:    # via pytest-cov
```

That said, I dug deeper into testing this and it turned out that what I described in https://github.com/containerbuildsystem/cachi2/pull/481#discussion_r1519822031 happens also with `coveralls==3.2.0` which then downgrades `coverage` to v5.5 with the same error. Looks like this all boils down to the fact that `coveralls` is an abandonware now: https://github.com/TheKevJames/coveralls-python/issues/393 and we should ultimately look for a different solution if we wish to keep reporting the coverage online (do we?). Since the behaviour of `coveralls` installation wrt/ `coverage` seems to be consistently broken no matter what version I am okay if you proceed with dropping the version pin, but please do it in a separate commit with and explanation - I think I've provided enough material to reference in the commit message :) .

Reply:
> if we wish to keep reporting the coverage online

IMO, it's more about having *some* report, vs. having to dig around in GH action logs (I'm still trying to get over the fact that we don't have useful reporting from pytest... (pytest-html, anyone?))

Reply:
Should I fill out a story on JIRA ?

Reply:
> Should I fill out a story on JIRA ?

I think we can keep this one contained solely in the upstream issue tracker rather than internally, but yes, we should track the coveralls issue, so that we don't forget about it.

Reply:
linking it here just for completeness: https://github.com/containerbuildsystem/cachi2/issues/498